guide_drafts/model-derives.md

```diff
@@ -0,0 +1,769 @@
+
+### Example 
+
+*Note: src/lib.rs with #[macro_use] extern crate diesel_codegen; not shown*
+
+*src/models.rs*
+```rust
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+#[derive(Insertable)]
+#[table_name="users"]
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    #[column_name(email)]
+    pub electronic_mail: &'a str,
+ }
+```
+
+*src/main.rs*
+```rust
+// externs, database connection, and using statement code omitted
+
+fn main() {
+  let new_user = NewUser { 
+      first_name: "Gordon", 
+      last_name: "Freeman", 
+      electronic_mail: "gordon.freeman@blackmesa.co" 
+  };
+
+  diesel::insert(&new_user).into(users::table)
+      .execute(&db_connection)
+      .expect("Error inserting row into database");
+
+  let all_users = users.load::<User>(&db_connection)
+      .expect("Error loading users");
+
+  println!("User count: {}", all_users.len()); // > User count: 1
+}
+```
+
+If we try to insert records without deriving `Insertable`, we would get the following error.
+
+```rust
+error[E0277]: the trait bound `&diesel_demo::models::NewUser<'_>: diesel::query_builder::insert_statement::IntoInsertStatement<_, diesel::query_builder::insert_statement::Insert>` is not satisfied
+  --> src/bin/main.rs:31:30
+   |
+31 |     diesel::insert(&new_user).into(users::table)
+   |                              ^^^^ the trait `diesel::query_builder::insert_statement::IntoInsertStatement<_, diesel::query_builder::insert_statement::Insert>` is not implemented for `&diesel_demo_cli::models::NewUser<'_>`
+```
+In the above error, the compiler is telling us there is an unsatisfied trait bound from IntoInsertStatement. `diesel::insert()` returns an `IncompleteInsertStatement`, which implements `into()`. `into()` specifies the table the `Insertable` data should be passed to, which is why this constraint is reported to us.
+
+The example below is how we might implement `Insertable` without the derive annotation. In this case, we are using the `impl_Insertable!` macro, but there are a few more ways which enable  us from having to reimplement our `NewUser` struct.
```

Comment:
> enable  us

double space

Also… this whole sentence is confusing. And using the macro is cheating! 😝